pvlib/tests/test_shading.py

```diff
@@ -327,3 +327,57 @@ def test_shaded_fraction1d_unprovided_shading_row_rotation():
     premises = test_data.drop(columns=["expected_sf"])
     sf = shading.shaded_fraction1d(**premises)
     assert_allclose(sf, expected_sf, atol=1e-2)
+
+
+@pytest.fixture
+def direct_martinez_Table2():
```

Comment:
Can we add a few edge case tests also?  For example with zero irradiance, zero shading, etc